bundles/framework/layerlist/view/LayerList/LayerCollapse/util.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * If both layers have same group, they are ordered by layer.getName()
+ * @param {Oskari.mapframework.domain.WmsLayer/Oskari.mapframework.domain.WfsLayer/Oskari.mapframework.domain.VectorLayer/Object} a comparable layer 1
+ * @param {Oskari.mapframework.domain.WmsLayer/Oskari.mapframework.domain.WfsLayer/Oskari.mapframework.domain.VectorLayer/Object} b comparable layer 2
+ * @param {String} method layer method name to sort by
+ */
+const comparator = (a, b, method) => {
+    var nameA = a[method]().toLowerCase();
+    var nameB = b[method]().toLowerCase();
+    if (nameA === nameB && (a.getName() && b.getName())) {
+        nameA = a.getName().toLowerCase();
+        nameB = b.getName().toLowerCase();
+    }
+    if (nameA < nameB) {
```

Comment:
We could use Oskari.util.naturalSort(nameA, nameB) here after we have determined what we need to compare instead of a simple comparison.

Reply:
Using natural sort